app/server.ml

```diff
@@ -346,16 +346,22 @@ let worker_loop t addr fd =
   | Ok Builder.Job_requested platform ->
     begin
       Logs.app (fun m -> m "job requested for %S" platform);
-      let rec find_job () =
-        let queue = find_queue t platform in
-        match Queue.take_opt queue with
-        | None ->
-          if not (template_exists t.cfgdir platform) then
-            Logs.warn (fun m -> m "no template for %S" platform);
+      let rec find_job = function
+        | `Platform_removed p ->
+          if String.equal p platform then
+            Logs.warn (fun m -> m "Platform %s removed; remember to shut down worker at %a"
+                          platform pp_sockaddr addr);
           Lwt_condition.wait t.waiter >>= find_job
-        | Some job -> Lwt.return job
+        | `New_job ->
+          let queue = find_queue t platform in
+          match Queue.take_opt queue with
+          | None ->
+            if not (template_exists t.cfgdir platform) then
+              Logs.warn (fun m -> m "no template for %S" platform);
+            Lwt_condition.wait t.waiter >>= find_job
+          | Some job -> Lwt.return job
       in
-      find_job () >>= fun job ->
+      Lwt_condition.wait t.waiter >>= find_job >>= fun job ->
```

Comment:
On second thought we probably need to trigger `` `New_job `` on first call to ensure queues are created(?)
```suggestion
      find_job `New_job >>= fun job ->
```